internal/database/postgres/connect.go

```diff
@@ -0,0 +1,39 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"github.com/dilyara4949/flight-booking-api/internal/config"
+	_ "github.com/lib/pq"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+func Connect(ctx context.Context, cfg config.Postgres) (*gorm.DB, error) {
+	dsn := fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable",
+		cfg.Host, cfg.Port, cfg.User, cfg.Password, cfg.DB)
+
+	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
+	if err != nil {
+		return nil, err
+	}
+
+	sqlDB, err := db.DB()
+	if err != nil {
+		return nil, err
+	}
+
+	sqlDB.SetMaxOpenConns(cfg.MaxConnections)
+
+	err = sqlDB.PingContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = Migrate(db)
+	if err != nil {
+		return nil, fmt.Errorf("	migration failed: %w", err)
+	}
```

Comment:
any reason for getting underlying driver?

Reply:
to ping connection and set maxconnection